firedrake/mesh.py

```diff
@@ -1517,77 +1554,82 @@ class VertexOnlyMeshTopology(AbstractMeshTopology):
     """
 
     @PETSc.Log.EventDecorator()
-    def __init__(self, swarm, parentmesh, name, reorder, use_cell_dm_marking, tolerance=0.5):
-        """
-        Half-initialise a mesh topology.
-
-        :arg swarm: Particle In Cell (PIC) DMSwarm representing
-            vertices immersed within a DMPlex stored in the
-            ``parentmesh`` argument.
-        :arg parentmesh: the mesh within which the vertex-only mesh
-            topology is immersed.
-        :arg name: name of the mesh
-        :arg reorder: whether to reorder the mesh (bool)
-        :arg use_cell_dm_marking: whether to inherit vertex markings
-            ("pyop2_core", "pyop2_owned" and "pyop2_ghost") from the cell DM,
-            i.e. the parent mesh DMPlex or DMSwarm. If true, this generally
-            means marking each vertex with the same label as the cell it
-            resides in. For extruded meshes this is the base mesh cell. If
-            false, we let mark_entity_classes do the marking. In such a case
-            we should not let there be any vertices in the halo regions.
-        :tolerance: The relative tolerance (i.e. as defined on the
-            reference cell) for the distance a point can be from a cell and
-            still be considered to be in the cell.
-        """
-
-        super().__init__(name, tolerance=tolerance)
-
-        # TODO: As a performance optimisation, we should renumber the
-        # swarm to in parent-cell order so that we traverse efficiently.
-        if reorder:
-            raise NotImplementedError("Mesh reordering not implemented for vertex only meshes yet.")
-
-        dmcommon.validate_mesh(swarm)
-        swarm.setFromOptions()
-
-        self._parent_mesh = parentmesh
-        self.topology_dm = swarm
-        r"The PETSc DM representation of the mesh topology."
+    def __init__(self, swarm, parentmesh, name, reorder, input_ordering_swarm=None, perm_is=None, distribution_name=None, permutation_name=None):
+        """Initialise a mesh topology.
+
+        Parameters
+        ----------
+        swarm : PETSc.DMSwarm
+            `PETSc.DMSwarm` representing Particle In Cell (PIC) vertices
+            immersed within a `PETSc.DM` stored in the ``parentmesh``.
+        parentmesh : PETSc.DM
```

Comment:
```suggestion
        parentmesh : AbstractMeshTopology
```
I think

Reply:
`parentmesh` should probably also be renamed `parentmeshtopology`

Reply:
I do not think we need to rename. If this class is `VertexOnlyMeshTopology`, the parent ought to be topology.

Either way, this is the term used in current `master`, so we should discuss this elsewhere if we need to.